applications/crossbar/src/modules/cb_cdrs.erl

```diff
@@ -360,9 +360,14 @@ load_chunked_cdr_ids(Context, RespType, Db, Ids) ->
     of
         'false' -> Context;
         {'ok', Results} ->
-            JObjs = [kz_json:get_value(<<"doc">>, Result)
+            HasFilters = crossbar_filter:is_defined(Context),
```

Comment:
You don't need to call it directly here, `crossbar_filter:by_doc/2` is calling it.

Reply:
Well, the idea is that we don't need to compute is_defined for every doc. So we can short-circuit calling by_doc because we already know if filters are defined (which is the first thing by_doc does).

Reply:
`is_defined` is doing the _short-circuit_ itself: https://github.com/2600hz/kazoo/blob/edocify/applications/crossbar/src/crossbar_filter.erl#L63

Reply:
By why call the function X times (where X could be large depending on pagination settings) when calling it once outside the loop and then doing the boolean test is available?